src/ruby/spec/symbol_spec.rb

```diff
@@ -0,0 +1,107 @@
+                    end
+                end
+            end
+            describe '#-' do
+                context 'with another Symbol object as argument' do
+                    it 'returns a Symbol object that is result of
+                        argument subtracted from self' do
+                        c = @a-@b
+                        expect(c).to be_an_instance_of SymEngine::Symbol
+                        expect(c.to_s).to eql("x - y")
+                    end
+                end
+            end
+            describe '#*' do
+                context 'with another Symbol object as argument' do
+                    it 'returns a Symbol object that is result of
+                        self multiplied by the argument' do
+                        c = @a*@b
+                        expect(c).to be_an_instance_of SymEngine::Symbol
+                        expect(c.to_s).to eql("x*y")
+                    end
+                end
+            end
+            describe '#/' do
+                context 'with another Symbol object as argument' do
+                    it 'returns a Symbol object that is result of
+                        self divided by the argument' do
+                        c = @a/@b
+                        expect(c).to be_an_instance_of SymEngine::Symbol
+                        expect(c.to_s).to eql("x/y")
+                    end
+                end
+            end
+            describe '#**' do
+                context 'with another Symbol object as argument' do
+                    it 'returns a Symbol object that is result of
+                        self raised to the power of argument' do
+                        c = @a**@b
+                        expect(c).to be_an_instance_of SymEngine::Symbol
+                        expect(c.to_s).to eql("x**y")
+                    end
+                end
+            end
+        end
+
+        describe 'unary operations' do
+            before :each do
+                @x = SymEngine::Symbol.new("a")
+            end
+            describe '#-@' do
+                context "doesn't take any argument" do
+                    it 'returns the negation of self' do
+                        p = -@x
+                        expect(p).to be_an_instance_of SymEngine::Symbol
+                        expect(p.to_s).to eql("-a")
+                    end
+                end
+            end
+        end
+
+        describe '#expand' do
+            context 'with nothing as argument' do
+                it 'returns the expanded form' do
+                    x = SymEngine::Symbol.new("x")
+                    y = SymEngine::Symbol.new("y")
+                    z = SymEngine::Symbol.new("z")
+                    e = (x+y+z)*(x+y+z)
+                    f = e.expand()
+                    expect(e.to_s).to eql("(x + y + z)**2")
+                    expect(f).to be_an_instance_of SymEngine::Symbol
+                    expect(f.to_s).to eql("2*x*y + 2*x*z + 2*y*z + x**2 + y**2 + z**2")
```

Comment:
In SymPy that's called simplify (f-g) == 0. We do not have simplify yet in
SymEngine until we implement more advanced polynomial algorithms.

Sent from my mobile phone.
On Jun 25, 2015 9:09 AM, "Abinash Meher" <notifications@github.com> wrote:

> In src/ruby/spec/symbol_spec.rb
> <https://github.com/sympy/symengine/pull/480#discussion_r33265382>:
>
> > +                    end
> > +                end
> > +            end
> > +        end
> > +
> > +        describe '#expand' do
> > +            context 'with nothing as argument' do
> > +                it 'returns the expanded form' do
> > +                    x = SymEngine::Symbol.new("x")
> > +                    y = SymEngine::Symbol.new("y")
> > +                    z = SymEngine::Symbol.new("z")
> > +                    e = (x+y+z)*(x+y+z)
> > +                    f = e.expand()
> > +                    expect(e.to_s).to eql("(x + y + z)**2")
> > +                    expect(f).to be_an_instance_of SymEngine::Symbol
> > +                    expect(f.to_s).to eql("2*x*y + 2*x*z + 2*y*z + x**2 + y**2 + z**2")
>
> Is there a method to check mathematical equality too? That will be nice to
> have in this minimal set of functions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/symengine/pull/480/files#r33265382>.
>